### PR TITLE
Switch profiler sync to use fast dispatch

### PR DESCRIPTION
**Pull request by mo-tenstorrent**
### Ticket
#13583 

### Problem description
Force SD dispatch is not supported anymore. 

### What's changed
Profiler sync uses fast dispatch to push the sync program

### Checklist
- [x] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/11348276715
- [x] T3K Profiler https://github.com/tenstorrent/tt-metal/actions/runs/11366706507
- [x] Device perf https://github.com/tenstorrent/tt-metal/actions/runs/11296316009
- [x] Sweep tests https://github.com/tenstorrent/tt-metal/actions/runs/11296337152
- [x] uBenchmark https://github.com/tenstorrent/tt-metal/actions/runs/11296341340

### CODE REVIEW

File: tt_metal/impl/device/device_pool.hpp [hunk 1]
```diff
@@ -66,6 +66,7 @@ class DevicePool {
     std::unordered_map<uint32_t, uint32_t> device_to_core_map;
 
     void init_firmware_on_active_devices() const;
+    void init_profiler_devices() const ;
```
Comment (tt-aho) [inline tt_metal/impl/device/device_pool.hpp:69]:
```suggestion
    void init_profiler_devices() const;
```

File: tt_metal/tools/profiler/tt_metal_profiler.cpp [hunk 2]
```diff
@@ -122,7 +122,7 @@ void syncDeviceHost(Device *device, CoreCoord logical_core, std::shared_ptr<tt_m
     }
     constexpr bool wait_for_all_cores_done = false;
     constexpr bool force_slow_dispatch = true;
-    LaunchProgram(device, sync_program, wait_for_all_cores_done, force_slow_dispatch);
+    EnqueueProgram(device->command_queue(), *sync_program, false);
```
Comment (tt-aho) [inline tt_metal/tools/profiler/tt_metal_profiler.cpp:125]: Removed unused constexprs?

**Review by tt-aho — APPROVED** [review]

**Review by tt-asaigal — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]

File: tt_metal/tools/profiler/process_device_log.py [hunk 3]
```diff
@@ -243,6 +244,11 @@ def import_device_profile_log(logPath):
                     devicesData["devices"][chipID] = {
                         "cores": {core: {"riscs": {risc: {"timeseries": [(timerID, timeData, statData)]}}}}
                     }
+                if doOpsDetection and "zone_name" in timerID.keys() and "dispatch" in timerID["zone_name"].lower():
+                    logger.warning(
+                        "Dispatch core data is detected, stats across ops are skipped as op detection cannot work with dispatch data present"
```
Comment (dmakoviichuk-tt) [inline tt_metal/tools/profiler/process_device_log.py:249]: hard to understand what it means. Could you add more explanations(something actionable).

**Review by dmakoviichuk-tt — APPROVED** [review]
LGTM with one comment to fix

File: tt_metal/tools/profiler/process_device_log.py (hunk 3 above)
Comment (mo-tenstorrent) [reply tt_metal/tools/profiler/process_device_log.py:249]: How is "Dispatch core data is detected: stats across ops are skipped as op detection cannot work with dispatch data present. Please unset the TT_METAL_DEVICE_PROFILER_DISPATCH environment variable"

**Pull request merged**